src/function/table/system/attach_duckdb.cpp

```diff
@@ -0,0 +1,241 @@
+
+	auto result = make_unique<ExternalQueryFunctionData>();
+	result->file_name = input.inputs[0].GetValue<string>();
+	result->table = input.inputs[1].GetValue<string>();
+
+	result->database = make_unique<DuckDB>(result->file_name);
+	result->connection = make_unique<Connection>(*result->database);
+
+	result->tablerelation = result->connection->Table(result->table)->Execute();
+
+	for (idx_t i = 0; i < result->tablerelation->names.size(); i++) {
+		return_types.push_back(result->tablerelation->types[i]);
+		names.emplace_back(result->tablerelation->names[i]);
+	}
+	return move(result);
+}
+
+static void ExternalQueryFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	auto &data = (ExternalQueryFunctionData &)*data_p.bind_data;
+	if (data.finished) {
+		return;
+	}
+
+	if (data.rowiterator == nullptr) {
+		data.rowiterator = make_unique<QueryResultIterator>(data.tablerelation.get());
+	}
+
+	auto dconn = Connection(context.db->GetDatabase(context));
+
+	idx_t count = 0;
+
+	while ((data.rowiterator)->result != nullptr && count < STANDARD_VECTOR_SIZE) {
+		for (idx_t col = 0; col < data.tablerelation->types.size(); col++) {
+			output.SetValue(col, count, (*(*(data.rowiterator))).GetValue(col));
+		}
+		count++;
+		data.rowiterator->Next();
+	}
+	if ((data.rowiterator)->result == nullptr) {
+		data.finished = true;
+	}
+	// todo copy views
+	// {
+	// 	check_ok(sqlite3_prepare_v2(db, "SELECT sql FROM sqlite_master WHERE type='view'", -1, &res, nullptr), db);
+
+	// 	while (sqlite3_step(res) == SQLITE_ROW) {
+	// 		auto view_sql = string((const char *)sqlite3_column_text(res, 0));
+	// 		dconn.Query(view_sql);
+	// 	}
+	// 	check_ok(sqlite3_finalize(res), db);
+	// }
+	output.SetCardinality(count);
+}
+
+struct AttachFunctionData : public TableFunctionData {
+	AttachFunctionData() {
+	}
+
+	bool finished = false;
+	bool overwrite = false;
+	bool temporary = false;
+	string file_name = "";
+};
+
+static unique_ptr<FunctionData> AttachBind(ClientContext &context, TableFunctionBindInput &input,
+                                           vector<LogicalType> &return_types, vector<string> &names) {
+
+	auto result = make_unique<AttachFunctionData>();
+	result->file_name = input.inputs[0].GetValue<string>();
+
+	for (auto &kv : input.named_parameters) {
+		if (kv.first == "overwrite") {
+			result->overwrite = BooleanValue::Get(kv.second);
+		}
+		if (kv.first == "temporary") {
+			result->temporary = BooleanValue::Get(kv.second);
+		}
+	}
+
+	return_types.push_back(LogicalType::BOOLEAN);
+	names.emplace_back("Success");
+	return move(result);
+}
+
+static void AttachFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	auto &data = (AttachFunctionData &)*data_p.bind_data;
+	if (data.finished) {
+		return;
+	}
+
+	DuckDB db {data.file_name};
+	auto econn = Connection(db);
+	auto dconn = Connection(context.db->GetDatabase(context));
```

Comment:
We shouldn't be opening a new connection to the database, we need to use the existing client context that is passed in (`ClientContext`), otherwise our transaction context does not carry over.